src/app/Http/Controllers/API/PollController.php

```diff
@@ -25,9 +28,16 @@ class PollController extends Controller
      *
      * @param Presentation $presentation
      */
-    public function index(Presentation $presentation)
+    public function index(Request $request, Presentation $presentation)
     {
-        //
+        $res = PollResource::collection($presentation->polls);
+
+        // filter drafts for viewers
+        if ($res->count() > 0 && $request->user()->id != $res->first()->presenter()) {
+            $res = $res->diff(Poll::whereIn('state', State::DRAFT())->get());
+        }
+
+        return $res;
```

Comment:
for that logic you can create a function in Presentation model named for example `nonDraftPolls`, using eloquent ORM that is more performant than collection parsing.

Reply:
Fixed in c7f492e9bf4c825a1215acb989a228f0951a5918